pkg/apis/integreatly/v1alpha1/grafana_types.go

```diff
@@ -29,6 +29,7 @@ type GrafanaSpec struct {
 	ServiceAccount         *GrafanaServiceAccount   `json:"serviceAccount,omitempty"`
 	Client                 *GrafanaClient           `json:"client,omitempty"`
 	Compat                 *GrafanaCompat           `json:"compat"`
+	DBPasswordRef          map[string]string        `json:"dbpasswordref,omitempty"`
```

Comment:
What about using v13.SecretKeySelector instead of map string string ? is there not posible i didn't got it

Reply:
I think it is not appliable

Reply:
For me it seems more reasonable due to we will have strict schema and doesn't allow consumer specify invalid keys, i think this better that use map where you can have type in key and in this case kubernetes can't check this

Reply:
How will this approach prevent invalid keys specifying?

Reply:
I mean what stop me from writing "secretReference" instead of "secretRef"

Reply:
How the proposed approach stop anybody to do that?

Reply:
Because in this case we can generate CRD with Open API scheme validation and kubernetes will validate all resources using custom resource validation all resources which not konform with schema will be rejected by kuberentes api